### DISCUSSION TITLE
Method of compile LibTorchSharp on ARM based Mac OS

### DISCUSSION THREAD

**Discussion by dayo05** [xamarin/TorchSharp#837] [Show and tell]
Do this on your risk!!

Before we go, please replace `path/to/TorchSharp` and `path/to/pytorch` to yours!
1. Build whole pytorch.
    A. Clone https://github.com/pytorch/pytorch.git
    B. install dependencies of compile. This is introduced on README.md on pytorch's github repo.
    C. Execute setup.py
2. Edit CMakeLists of LibTorchSharp
* Some feature is not exists on ARM. So, I just disable some features. 
    A. Open /path/to/TorchSharp/src/Native/LibTorchSharp/CMakeLists.txt
    B. Remove `crc32c.c` and `crc32c.h`. Those files uses cpuinfo.h which not exists on ARM.
3. Build LibTorchSharp
    A. Generate MakeFile with this command: `CMAKE_PREFIX_PATH=path/to/pytorch/torch/share/cmake Caffe2_DIR=path/to/pytorch/torch/share/cmake/Caffe2 cmake .`
    B. Build with command `make`

If this method works fine, native library will exists on path/to/TorchSharp/src/Native/LibTorchSharp/libTorchSharp.dylib

I don't test this yet but maybe you can load the library via `NativeLibrary.Load`

**Comment by dayo05** [comment]
How to test:
1. Edit TorchSharpTest.csproj: You should change target platform from `x64` to `arm64`.
2. Load Native Library on constructor. In my case, I did like this.
```cs
NativeLibrary.Load("/Users/dayo/TorchSharp/src/Native/pytorch/build/lib/libtorch.dylib");
NativeLibrary.Load("/Users/dayo/TorchSharp/src/Native/LibTorchSharp/libLibTorchSharp.dylib");        
```

I check that test passed.

**Comment by FancyFurret** [comment]
I have integrated an arm64 build into the build system using the arm64 torch binaries from anaconda: PR #903

If you're still interested in arm64 support, maybe you can help validate the PR? :)
